src/createMetricsModule.ts

```diff
@@ -0,0 +1,164 @@
+import type { Request, Response, NextFunction, Express } from 'express'
+import type { Gauge, Counter, Histogram } from 'prom-client'
+import type { Logger } from 'winston'
+
+const prometheus = require('prom-client')
+
+type DefaultMetricsLabels = {
+  containerImageTag: string
+  ecsClusterName: string
+  ecsServiceName: string
+  ecsTaskID: string
+  awsAccountName: string
+  instance: string
+  [key: string]: string
+}
+
+type CustomMetrics = {
+  [name: string]: Gauge | Counter | Histogram
+}
+
+export type MetricsConfigurationOptions = {
+  prefix?: string
+  defaultMetricsLabels: DefaultMetricsLabels
+}
+
+export type MetricsModule = {
+  registerCounter: (name: string, help: string, labelNames: string[]) => Counter
+  registerGauge: (name: string, help: string, labelNames: string[]) => Gauge
+  registerHistogram: (name: string, help: string, labelNames: string[], buckets?: number[]) => Histogram
+}
+
+const createMetricsModule = (app: Express, logger: Logger, options: MetricsConfigurationOptions): MetricsModule => {
```

Comment:
it could be but I thought it might be preferable to allow users to set additional default labels on top of the required ecs metadata which is why the type definition includes `[key: string]: string` in addition to the required keys

Reply:
Maybe you could keep the default labels hardcoded, and allow and additional labels to be passed in?